### DISCUSSION TITLE
Less-loader error compiling on v5 beta

### DISCUSSION THREAD

**Discussion by jeffvli** [rsuite/rsuite#1887] [Q&A] [answered]
I'm currently trying to upgrade my app to `rsuite@5.0.0-beta6` from the stable `4.10.2` version and I'm getting a compilation error that does not occur on v4. I'm using less-loader `^7.2.1` with less `3.7.0`, following the versions of the rsuite repository, along with the following options my webpack config. I've tried enabling/disabling a couple options including strictMath, but nothing seems to resolve the error for me.

```
× ｢wdm｣: ERROR in ./src/styles/App.global.css (./node_modules/css-loader/dist/cjs.js??ruleSet[1].rules[2].use[1]!./node_modules/less-loader/dist/cjs.js??ruleSet[1].rules[2].use[2]!./src/styles/App.global.css)
Module build failed (from ./node_modules/less-loader/dist/cjs.js):

.make-grid(xs);
^
Error evaluating function `each`: Operation on an invalid type
      Error in C:\code\app\node_modules\rsuite\Grid\styles\index.less (line 57, column 0)
 @ ./src/styles/App.global.css 47:4-60:5 2:12-194 9:17-24 13:7-21 45:20-34 49:6-59:7 50:38-52 56:26-40 58:21-28 68:15-29
 @ ./src/App.tsx 14:0-33
 @ ./src/index.tsx 11:0-24 36:36-39
```

```
{
  loader: 'less-loader',
  options: {
    lessOptions: {
      javascriptEnabled: true,
    },
  },
},
```

Is there a less option or something else that I need to bypass this error? Thank you.

**Comment by SevenOutman** [comment, marked as answer]
RSuite Less uses `range()` function which was introduced in Less 3.9.0. Try upgrade your Less compiler to a version no earlier than 3.9.0 and see if the error is resolved.

    npm install -D less@latest

**Reply by jeffvli** [reply]
Thank you that solved it! I'm not sure why I didn't try upgrading my Less version.
